Sherlock/pubsub.h

```diff
@@ -205,6 +222,9 @@ class PubSubHTTPEndpointImpl {
   // TODO(dkorolev): This is a long shot, but looks right: For type-filtered HTTP subscriptions,
   // whether we should terminate or no depends on `nowait`.
   ss::EntryResponse EntryResponseIfNoMorePassTypeFilter() const {
+    if (time_to_terminate_) {
+      return ss::EntryResponse::Done;
+    }
     return no_wait_ ? ss::EntryResponse::Done : ss::EntryResponse::More;
```

Comment:
Huh okay. The point of my style is the increased readability of return+ternary and assign+ternary (emphasize that the returned/assigned value is the whole parenthesized expression which is ternary inside).